x-pack/plugins/security_solution/public/timelines/components/timeline/data_providers/add_data_provider_popover.tsx

```diff
@@ -0,0 +1,194 @@
+          kqlQuery: '',
+          type,
+          queryMatch: {
+            displayField: undefined,
+            displayValue: undefined,
+            field,
+            value,
+            operator,
+          },
+          and: [],
+        },
+        timelineId,
+      });
+    },
+    [dataProviders, timelineId, dispatch, handleClosePopover]
+  );
+
+  const panels = useMemo(
+    () => [
+      {
+        id: 0,
+        width: 400,
+        items: [
+          {
+            name: ADD_FIELD_LABEL,
+            icon: <EuiIcon type="plusInCircle" size="m" />,
+            panel: 1,
+          },
+          timelineType === TimelineType.template
+            ? {
+                disabled: timelineType !== TimelineType.template,
+                name: ADD_TEMPLATE_FIELD_LABEL,
+                icon: <EuiIcon type="visText" size="m" />,
+                panel: 2,
+              }
+            : null,
+        ].filter((item) => item !== null) as EuiContextMenuPanelItemDescriptor[],
+      },
+      {
+        id: 1,
+        title: ADD_FIELD_LABEL,
+        width: 400,
+        content: (
+          <StatefulEditDataProvider
+            browserFields={browserFields!}
+            field=""
+            isExcluded={false}
+            onDataProviderEdited={handleDataProviderEdited}
+            operator=":"
+            timelineId={timelineId}
+            value=""
+            type={DataProviderType.default}
+          />
+        ),
+      },
+      {
+        id: 2,
+        title: ADD_TEMPLATE_FIELD_LABEL,
+        width: 400,
+        content: (
+          <StatefulEditDataProvider
+            browserFields={browserFields!}
+            field=""
+            isExcluded={false}
+            onDataProviderEdited={handleDataProviderEdited}
+            operator=":"
+            timelineId={timelineId}
+            value=""
+            type={DataProviderType.template}
+          />
+        ),
+      },
+    ],
+    [browserFields, handleDataProviderEdited, timelineId, timelineType]
+  );
+
+  const button = useMemo(
+    () => (
+      <EuiButton
+        size="s"
+        onClick={handleOpenPopover}
+        data-test-subj="addFilter"
+        iconType="arrowDown"
+        fill
+        iconSide="right"
+      >
+        <EuiText size="s">{ADD_FIELD_LABEL}</EuiText>
+      </EuiButton>
+    ),
+    [handleOpenPopover]
+  );
+
+  const content = useMemo(() => {
+    if (timelineType === TimelineType.template) {
+      return <EuiContextMenu initialPanelId={0} panels={panels} />;
+    }
+
+    return (
+      <StatefulEditDataProvider
+        browserFields={browserFields!}
+        field=""
+        isExcluded={false}
+        onDataProviderEdited={handleDataProviderEdited}
+        operator=":"
+        timelineId={timelineId}
+        value=""
+        type={DataProviderType.default}
+      />
+    );
+  }, [browserFields, handleDataProviderEdited, panels, timelineId, timelineType]);
+
+  return (
+    <EuiFlexItem grow={false}>
```

Comment:
with this comment [here](https://github.com/elastic/kibana/pull/67837/files#r450222055), I am wondering if you really need a EuiFlexItem. And also here `x-pack/plugins/security_solution/public/timelines/components/timeline/data_providers/providers.tsx` line 164. Does not look like that it needs to be wrap under a `EuiFlexItem`